### PR TITLE
rbd: rados namespaced image deletion fix

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixing an issue regarding namespaced rbd images.

Related issues
Fixes: #1397
Fixes: #1398

### CODE REVIEW

**Review by nixpanic — DISMISSED** [review]

File: internal/rbd/clone.go [hunk 1]
```diff
@@ -134,6 +134,9 @@ func (rv *rbdVolume) generateTempClone() *rbdVolume {
 	tempClone.ClusterID = rv.ClusterID
 	tempClone.Monitors = rv.Monitors
 	tempClone.Pool = rv.Pool
+	if rv.RadosNamespace != "" {
```
Comment (humblec) [inline internal/rbd/clone.go:137]: @Madhu-1 do we need this check ? if its not set on `rv.RadosNamespace` , its any way unset .

Comment (Madhu-1) [reply internal/rbd/clone.go:137]: Yes, we need this check as we are generating a new temp volume structure from rv struct?

Comment (Madhu-1) [reply internal/rbd/clone.go:137]: This commit was picked from the PR, keeping it wont be harmful

Comment (humblec) [reply internal/rbd/clone.go:137]: well, there is no reason to have an extra check.

Comment (Madhu-1) [reply internal/rbd/clone.go]: updated now PTAL

**Review by nixpanic — APPROVED** [review]
@mergifyio rebase

**Comment by nixpanic** [comment]
@Mergifyio rebase

**Comment by nixpanic** [comment]
**DCO** failed, probably due to the double space in the `Signed-off-by: Marton  Natko`. I'll approve it, but in case you want to correct it, feel free.

**Comment by Madhu-1** [comment]
> **DCO** failed, probably due to the double space in the `Signed-off-by: Marton Natko`. I'll approve it, but in case you want to correct it, feel free.

done @nixpanic

**Review by nixpanic — APPROVED** [review]

**Comment by nixpanic** [comment]
@Mergifyio rebase

**Comment by nixpanic** [comment]
@humblec please have a look, this is a simple bugfix that should get backported to v3.1

**Review by humblec — APPROVED** [review]

**Comment by mykaul** [comment]
Do we have a test in CI for this?

**Comment by Madhu-1** [comment]
> Do we have a test in CI for this?

Yes @mykaul added backend validation https://github.com/ceph/ceph-csi/pull/1501/commits/28c70b551cc3d93433a8e3420b5db735add4e327 for the fix

**Comment by discostur** [comment]
@humblec @Madhu-1 shouldn't this get backported to v3.1? Because v3.1.2 was already released but without that bugfix ...

**Comment by Madhu-1** [comment]
@discostur its backported to v3.1 branch in https://github.com/ceph/ceph-csi/pull/1527

**Pull request merged**